### PR TITLE
BUILD-2100 Use Vault

### PR DESCRIPTION
**Pull request by elrzn**
- [x] `.cirrus.yml`
- [x] `.github/workflows/release.yml`

### CODE REVIEW

File: .cirrus.yml [hunk 1]
```diff
@@ -30,11 +33,11 @@ build_task:
     cpu: 2
     memory: 2G
   env:
-    SIGN_KEY: ENCRYPTED[!cc216dfe592f79db8006f2a591f8f98b40aa2b078e92025623594976fd32f6864c1e6b6ba74b50647f608e2418e6c336!]
-    PGP_PASSPHRASE: ENCRYPTED[!314a8fc344f45e462dd5e8dccd741d7562283a825e78ebca27d4ae9db8e65ce618e7f6aece386b2782a5abe5171467bd!]
-    GITHUB_TOKEN: ENCRYPTED[!f458126aa9ed2ac526f220c5acb51dd9cc255726b34761a56fc78d4294c11089502a882888cef0ca7dd4085e72e611a5!]
+    SIGN_KEY: VAULT[development/kv/data/sign data.key]
+    PGP_PASSPHRASE: VAULT[development/kv/data/sign data.passphrase]
+    GITHUB_TOKEN: VAULT[development/github/token/SonarSource-sonar-lits-ro token]
```
Comment (Godin) [inline .cirrus.yml:38]: This `GITHUB_TOKEN` seems useless.

**Review by Godin — CHANGES_REQUESTED** [review]

**Review by Godin — CHANGES_REQUESTED** [review]
`.github/workflows/release.yml` should also be updated.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -9,72 +9,11 @@ env:
...
-        env:
-          SLACK_WEBHOOK_URL: ${{ secrets.SLACK_BUILD_WEBHOOK }}
+  release:
+    permissions:
+      id-token: write
+      contents: write
+    uses: SonarSource/gh-action_release/.github/workflows/main.yaml@5.0.1
+    with:
+      publishToBinaries: true
+      mavenCentralSync: true
```
Comment (elrzn) [inline .github/workflows/release.yml:19]: Note: No Slack notifications configured as I don't know which channel to use.

File: .cirrus.yml [hunk 3]
```diff
@@ -2,18 +2,21 @@ gcp_credentials: ENCRYPTED[!149d4005ecdba4cdd78bb5ba22756ebb98bf8e3367ee2e9ab08c
...
-  ARTIFACTORY_PRIVATE_USERNAME: repox-private-reader-lt-1a7e1f
-  ARTIFACTORY_PRIVATE_PASSWORD: ENCRYPTED[!4890acae4038fb09d3921b1126aad6af4c0bc3984e603cf1639766e44cc987c3f2b7d529b5420b2e3eca0f354b938bf8!]
-  ARTIFACTORY_DEPLOY_USERNAME: repox-qa-deployer-lt-1a7e1f
-  ARTIFACTORY_DEPLOY_PASSWORD: ENCRYPTED[!91fd8560ac00c4661c3161af1bd6e74ed8de8799e9d6ddc2f07bbcf154703adfb54d197ec2286c25f481fca7aba18c76!]
+  CIRRUS_VAULT_AUTH_PATH: jwt-cirrusci
+  CIRRUS_VAULT_ROLE: cirrusci-${CIRRUS_REPO_OWNER}-${CIRRUS_REPO_NAME}
+  CIRRUS_VAULT_URL: https://vault.sonar.build:8200
+  ARTIFACTORY_URL: VAULT[development/kv/data/repox data.url]
+  ARTIFACTORY_PRIVATE_USERNAME: vault-${CIRRUS_REPO_OWNER}-${CIRRUS_REPO_NAME}-private-reader
+  ARTIFACTORY_PRIVATE_PASSWORD: VAULT[development/artifactory/token/SonarSource-sonar-lits-private-reader access_token]
```
Comment (Godin) [inline .cirrus.yml:10]:
```suggestion
  ARTIFACTORY_PRIVATE_PASSWORD: VAULT[development/artifactory/token/${CIRRUS_REPO_OWNER}-${CIRRUS_REPO_NAME}-private-reader access_token]
```

File: .cirrus.yml [hunk 4]
```diff
@@ -2,18 +2,21 @@ gcp_credentials: ENCRYPTED[!149d4005ecdba4cdd78bb5ba22756ebb98bf8e3367ee2e9ab08c
...
+  ARTIFACTORY_DEPLOY_USERNAME: vault-${CIRRUS_REPO_OWNER}-${CIRRUS_REPO_NAME}-qa-deployer
+  ARTIFACTORY_DEPLOY_PASSWORD: VAULT[development/artifactory/token/SonarSource-sonar-lits-qa-deployer access_token]
```
Comment (Godin) [inline .cirrus.yml:12]:
```suggestion
  ARTIFACTORY_DEPLOY_PASSWORD: VAULT[development/artifactory/token/${CIRRUS_REPO_OWNER}-${CIRRUS_REPO_NAME}-qa-deployer access_token]
```

File: .cirrus.yml [hunk 5]
```diff
@@ -2,18 +2,21 @@ gcp_credentials: ENCRYPTED[!149d4005ecdba4cdd78bb5ba22756ebb98bf8e3367ee2e9ab08c
...
   #Possible values for ARTIFACTORY_DEPLOY_REPO: sonarsource-private-qa, sonarsource-public-qa
   ARTIFACTORY_DEPLOY_REPO: sonarsource-public-qa
-  ARTIFACTORY_API_KEY: ENCRYPTED[!4890acae4038fb09d3921b1126aad6af4c0bc3984e603cf1639766e44cc987c3f2b7d529b5420b2e3eca0f354b938bf8!]
+  ARTIFACTORY_ACCESS_TOKEN: VAULT[development/artifactory/token/SonarSource-sonar-lits-private-reader access_token]
```
Comment (Godin) [inline .cirrus.yml:15]:
```suggestion
  ARTIFACTORY_ACCESS_TOKEN: VAULT[development/artifactory/token/${CIRRUS_REPO_OWNER}-${CIRRUS_REPO_NAME}-private-reader access_token]
```

File: .cirrus.yml [hunk 6]
```diff
@@ -76,8 +78,9 @@ promote_task:
     memory: 500M
   env:
     #promotion cloud function
-    GCF_ACCESS_TOKEN: ENCRYPTED[!1fb91961a5c01e06e38834e55755231d649dc62eca354593105af9f9d643d701ae4539ab6a8021278b8d9348ae2ce8be!]
-    PROMOTE_URL: ENCRYPTED[!e22ed2e34a8f7a1aea5cff653585429bbd3d5151e7201022140218f9c5d620069ec2388f14f83971e3fd726215bc0f5e!]
+    GCF_ACCESS_TOKEN: VAULT[development/kv/data/promote data.token]
+    PROMOTE_URL: VAULT[development/kv/data/promote data.url]
+    GITHUB_TOKEN: VAULT[development/github/token/SonarSource-sonar-lits-promotion token]
```
Comment (Godin) [inline .cirrus.yml:83]:
```suggestion
    GITHUB_TOKEN: VAULT[development/github/token/${CIRRUS_REPO_OWNER}-${CIRRUS_REPO_NAME}-promotion token]
```

**Review by Godin — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml (hunk 2 above)
Comment (Godin) [reply .github/workflows/release.yml:19]: @elrzn added

**Review by Godin — APPROVED** [review]

**Pull request merged**